proto/src/determined/trial/v1/trial.proto

```diff
@@ -284,6 +285,8 @@ message MetricsReport {
   int32 id = 6;
   // Run ID of trial when metric was reported.
   int32 trial_run_id = 7;
+  // Run ID of trial when metric was reported.
+  string group = 8;
```

Comment:
are these datapoints going to be dumping a lot of data?

if so, `MetricsReport` could be inefficient, because 
- it includes `trial_id`, `group` on each data point, which is somewhat redundant
- `archived` flag which is useless because it'll always be false
- `id` which is useless because it can't be used in any way
- `trial_run_id`: I am not sure how end-user could use it 

For high-volume metrics view we use `ComparableTrial` + `DownsampledMetrics` https://github.com/determined-ai/determined/blob/main/proto/src/determined/api/v1/trial.proto#L408
The name is somewhat awkward, it doesn't really have anything to do with "downsampling", but the idea is to have metadata separately.

Reply:
So, I agree with the notion that this `MetricsReport` object is a bad representation of the data as it carries way too much metadata on every point. One issue I have with the idea of refactoring this though is that all of our `get_<group>_metrics()` functions return it. Are we planning on refactoring away from that format entirely? I wouldn't want to only support these "downsampled" metrics on the new endpoints that I'm providing if that wasn't the direction we actually wanted to go.

Is this the right PR to make a change like that? It seems like a very hairy issue that may require it's own PR from what I understand. I don't know how many things would be affected by a change like that.

To the question of:
> are these datapoints going to be dumping a lot of data?

The thing is that an inference use case, as far as I understand it, will likely not involve a lot of metrics. The user will likely want a handful of values for metrics over an entire run, rather than the running time-series format that our data currently uses.

I imagine that a lot of the metrics users would want would end up being custom metrics to determine "how good is this model?" rather than our constant time-series feedback loop format where the metrics are designed to feedback into the system for the training or validation process.

Reply:
we've touch this when we talked a few days back. I agree this is too much for this PR, please ignore this comment.

Reply:
Ticket for this: https://hpe-aiatscale.atlassian.net/browse/MLG-877